### ISSUE TITLE
Cannot mix NumPy dtypes float64 and float32 

### ISSUE THREAD

**Issue by farazk86** [ludwig-ai/ludwig#2223]
**Describe the bug**

I have a text dataset where each row has multiple labels. I am trying to perform multi-label classification on it. Cant share the dataset but has the following simple structure
```
| both_text      | labels                               |
|----------------|--------------------------------------|
| Free text here | economy health labour                |
| Free text here | health lawcrime tactical             |
| Free text here | economy health education environment |
```

**To Reproduce**

I'm using the following config file to train my model

```yaml
config_transformer = {
    "input_features": [
        {
            "name": "both_text",
            "type": "text",
            "preprocessing": {
                "tokenizer": "space",
            },
            "encoder": "transformer",
            "num_layers": 3,  # the number of transformer blocks
            "reduce_output": None,
        }
    ],
    "preprocessing": {
        "split_probabilities": [0.8, 0.1, 0.1],
    },
    "output_features": [
        {
            "name": "labels",
            "type": "set",
        },
    ],
    "trainer": {
        "epochs": 50,
        "early_stop": 3,
        "batch_size": 128,
    }

}
```

The below code to train and evaluate

```python
train_stats, preprocessed_data, train_output_directory = model.train(dataset=train_df, experiment_name=experiment)

# evaluate the model and save the results
test_stats, predictions, test_output_directory = model.evaluate(
  test_df,
  collect_predictions=True,
  collect_overall_stats=True,
  output_directory=os.path.join('results', experiment + '_run', 'pred_results'),
  skip_save_predictions=False,
  skip_save_eval_stats=False,
)
```

The training goes ahead fine but at the evaluation stage I get the following error:

```
Cannot mix NumPy dtypes float64 and float32', 'Conversion failed for column labels_probabilities with type object
```

Doing a Google search on that led me to this SO answer: https://stackoverflow.com/questions/67502942/arrowtypeerror-did-not-pass-numpy-dtype-object-conversion-failed-for-column

I did as advised and downgraded my numpy model to ``1.19.5`` and that seems to fix the problem and now my prediction output generates.

Dont know how this will affect the rest of Ludwig, but just wanted to bring this to the attention of the devs :)

**Environment (please complete the following information):**

- OS: \[e.g. iOS\] Windows 10
- Version \[e.g. 22\] 64 bit
- Python version 3.8.10
- Ludwig version 0.5.3

**Comment by farazk86** [comment]
Update... 

So now all of a sudden even the downgraded numpy version is not working..

**Comment by justinxzhao** [comment]
Hi @farazk86, heads up that this issue might be fixed by #2211, which fixes prediction saving for models with Set output. CC: @geoffreyangus
